addons/l10n_jo/models/account_journal.py

```diff
@@ -0,0 +1,21 @@
+from odoo import fields, models, api
+
+
+class AccountJournal(models.Model):
+    _inherit = 'account.journal'
+
+    @api.depends('company_id.account_journal_payment_credit_account_id')
+    def _compute_outbound_payment_method_line_ids(self):
+        super()._compute_outbound_payment_method_line_ids()
+        for journal in self:
+            if journal.country_code == 'JO' and journal.type in ('bank', 'cash'):
+                for line in journal.outbound_payment_method_line_ids:
+                    line.payment_account_id = journal.company_id.account_journal_payment_credit_account_id
```

Comment:
Could you give more insights into why this account should be different?  (maybe I should check the task)